crates/red_knot_python_semantic/src/types/diagnostic.rs

```diff
@@ -802,12 +802,8 @@ impl Diagnostic for TypeCheckDiagnostic {
         TypeCheckDiagnostic::message(self).into()
     }
 
-    fn file(&self) -> Option<File> {
-        Some(TypeCheckDiagnostic::file(self))
-    }
-
-    fn range(&self) -> Option<TextRange> {
-        Some(Ranged::range(self))
+    fn span(&self) -> Option<Span> {
```

Comment:
Same: I'm sort of inclined to simply store the `Span` directly on the `Diagnostic` struct. But I guess this way is fine too

Reply:
This one is a little trickier because a `TypedDiagnostic` always has a `File` and a `Range`, where as a `Span` always has a `File` and an optional `Range`. So I'm going to leave this as-is for now.